### PR TITLE
webpack optimization -- optimized Lodash build

### PR DESCRIPTION
**Pull request by EmanekaT**
## Summary
Changes
1. Changed import syntax to avoid importing the whole lodash module. Files in the test folders remain untouched.
2. ~Added lodash-webpack-plugin~ I will test it more before adding it. 

Comparison
https://docs.google.com/spreadsheets/d/13uYVbPvEB-LIBg_0CAx28Cex0mDWi03V1_4TVWb3G-0/edit?usp=sharing

### CODE REVIEW

File: frontend_build/src/parse_bundle_plugin.js [hunk 1]
```diff
@@ -130,6 +131,10 @@ var parseBundlePlugin = function(data, base_dir) {
       __publicPath: JSON.stringify(publicPath),
     }),
     new extract$trs(data.locale_data_folder, data.name),
+    new LodashModuleReplacementPlugin({
+      'collections': true,
+      'paths': true
```
Comment (rtibbles) [inline frontend_build/src/parse_bundle_plugin.js]: I think the linting error is probably coming from a lack of a trailing `,` here.

The pre-commit hooks are meant to catch these kind of things when you commit on your local machine: http://kolibri.readthedocs.io/en/develop/dev/getting_started.html#pre-commit-install

To fix this issue for sure, run the following command in your kolibri directory:
`yarn run lint-js:fix -- frontend_build/src/parse_bundle_plugin.js`.

File: frontend_build/src/parse_bundle_plugin.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ var extract$trs = require('./extract_$trs');
 var merge = require('webpack-merge');
 var ExtractTextPlugin = require('extract-text-webpack-plugin');
 var WebpackRTLPlugin = require('webpack-rtl-plugin');
+var LodashModuleReplacementPlugin = require('lodash-webpack-plugin');
```
Comment (rtibbles) [inline frontend_build/src/parse_bundle_plugin.js]: Wondering what difference this makes to the final build, if we have already been careful about all of our lodash imports?

Comment (mzq592) [reply frontend_build/src/parse_bundle_plugin.js]: This decreased sizes from 100-200kb to under 50kb. Its documentation says "it complements babel-plugin-lodash by shrinking its cherry-picked builds even further"!

File: frontend_build/src/parse_bundle_plugin.js (hunk 1 above)
Comment (rtibbles) [reply frontend_build/src/parse_bundle_plugin.js]: (sorry, updated command)

File: frontend_build/src/parse_bundle_plugin.js (hunk 2 above)
Comment (rtibbles) [reply frontend_build/src/parse_bundle_plugin.js]: Wow, do we have an idea of what this is actually doing under the hood? Just wondering what the price is for these huge savings!

Comment (mzq592) [reply frontend_build/src/parse_bundle_plugin.js]: I think it automatically removes the list of features here. https://github.com/lodash/lodash-webpack-plugin
We choose to keep features we need. I am not sure which features are used and which are not in kolibri though.

**Comment by rtibbles** [comment]
OK, just need to remove lodash from the yarn lock file here.

Make sure to remove it with `yarn remove lodash-webpack-plugin` or it will not remove it from the yarn lock file as well.

File: kolibri/plugins/document_pdf_render/assets/src/views/index.vue [hunk 3]
```diff
@@ -58,7 +58,8 @@
   import responsiveElement from 'kolibri.coreVue.mixins.responsiveElement';
   import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
   import { sessionTimeSpent } from 'kolibri.coreVue.vuex.getters';
-  import { debounce, throttle } from 'lodash';
```
Comment (DXCanas) [inline kolibri/plugins/document_pdf_render/assets/src/views/index.vue:61]: thanks guys :+1:

**Pull request merged**